cryptographic_estimators/UOVEstimator/UOVAlgorithms/collision_attack.py

```diff
@@ -0,0 +1,170 @@
+        alpha = self._alpha
+        r = self._gray_code_eval_cost
+        X = 9.25  +  (1 / 2) * (-log2(3 * m * r)  + log2(alpha) + m * log2(q))
+        precision = 2
+        X_truncated = float(int(X * 10 ** precision) / 10 ** precision)
+        return X_truncated
+    
+    @optimal_parameter
+    def Y(self):
+        """
+        Return logarithm of the optimal `Y`, i.e. logarithm of no. of hashes to compute
+
+        EXAMPLES::
+
+            sage: from cryptographic_estimators.UOVEstimator.UOVAlgorithms.collision_attack import CollisionAttack
+            sage: from cryptographic_estimators.UOVEstimator.uov_problem import UOVProblem
+            sage: E = CollisionAttack(UOVProblem(n=24, m=10, q=2))
+            sage: E.Y()
+            0.0
+
+        """
+        X =  self.X()
+        alpha = self._alpha
+        _, m, q = self.problem.get_parameters()
+        Y = max(log2(alpha) + m * log2(q) - X, 0)
+        precision = 2
+        Y_truncated = float(int(Y * 10 ** precision) / 10 ** precision)
+        return Y_truncated
+
+    def _compute_time_complexity(self, parameters: dict):
+        """
+        Return the time complexity of the algorithm for a given set of parameters
+
+        INPUT:
+
+        - ``parameters`` -- dictionary including the parameters
+
+        TESTS::
+
+            sage: from cryptographic_estimators.UOVEstimator.UOVAlgorithms.collision_attack import CollisionAttack
+            sage: from cryptographic_estimators.UOVEstimator.uov_problem import UOVProblem
+            sage: E = CollisionAttack(UOVProblem(n=24, m=10, q=2))
+            sage: E.time_complexity()
+            17.387737896811494
+
+        """
+        _, m, q = self.problem.get_parameters()
+        X = self.X()
+        Y = self.Y()
+        alpha = self._alpha
+        r = self._gray_code_eval_cost
+        time_temp = int((3 / 2) * m * r * 2 ** X) + int(2 ** 17.5 * 2 ** Y)
+        time_in_bits = log2(time_temp)
+        time_in_bits += log2(1/(1 - e ** (-alpha)))
+        cost_one_field_mult = self.problem.to_bitcomplexity_time(1)
+        time = time_in_bits - cost_one_field_mult
+        return time
+
+    def _compute_memory_complexity(self, parameters: dict):
+        """
+        Return the memory complexity of the algorithm for a given set of parameters
+
+        INPUT:
+
+        - ``parameters`` -- dictionary including the parameters
+
+        TESTS::
+
+            sage: from cryptographic_estimators.UOVEstimator.UOVAlgorithms.collision_attack import CollisionAttack
+            sage: from cryptographic_estimators.UOVEstimator.uov_problem import UOVProblem
+            sage: E = CollisionAttack(UOVProblem(n=24, m=10, q=2))
+            sage: E.memory_complexity()
+            8.0
+
+        """
+        X = self.X()
+        Y = self.Y()
+        _, m, q = self.problem.get_parameters()
+        mem_evals = log2(log2(q)) + log2(m) + X
+        mem_hashes = log2(256) + Y
+        return min(mem_evals, mem_hashes)
```

Comment:
shouldnt this be a  `max`?

Reply:
It should be the minimum since you only have to store either the hashes or polynomial evaluations.